mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,407 @@
+
+        rec.setClanDeaths(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_DEATHS);
+
+        this.clanDeaths = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds assists to a gg.solarmc.loader.clans.Clan.
+     * @param transaction tx
+     * @param toAdd the amount you want to add
+     * @return Accurate assists after transaction
+     */
+    public Integer addAssists(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanAssists() + toAdd;
+
+        rec.setClanAssists(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_ASSISTS);
+
+        this.clanAssists = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Returns the accurate allied clan to this object.
+     * @param transaction
+     * @return
+     */
+    public Optional<Clan> getAlliedClan(Transaction transaction) {
+        if (this.alliedClan == null) return Optional.empty();
+
+        return Optional.of(manager.getClan(transaction,alliedClan));
+    }
+
+    /**
+     * Gets all cached clan members
+     * @param transaction
+     * @return
+     */
+    public Set<ClanMember> getClanMembers(Transaction transaction) {
+        return transaction.getProperty(DSLContext.class)
+                .select(CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .from(CLANS_CLAN_MEMBERSHIP)
+                .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(this.clanID))
+                .fetchSet((rec) -> new ClanMember(this.clanID,rec.value1(),manager));
+    }
+
+    /**
+     * Adds a member to the clan. Note that this does not have a built in limit, API implementer needs to implement
+     * limit if we desire a limit on members
+     *
+     * Note to API implementer: This handles nothing regarding invites. If you want an invite system, it needs to come
+     * before this method.
+     *
+     * @param transaction The tx
+     * @param receiver gg.solarmc.loader.clans.ClanDataObject to add.
+     * @return Accurate set of ClanMembers post transaction
+     * @throws IllegalStateException if member is already a member of another clan
+     * @throws IllegalArgumentException if member is the leader of the clan
+     */
+    public Set<ClanMember> addClanMember(Transaction transaction, ClanDataObject receiver) {
+        if (receiver.isSimilar(leader)) throw new IllegalArgumentException("Tried to add leader as member");
+        if (receiver.getClan(transaction).isPresent()) throw new IllegalStateException("Object is already a member of another clan");
+
+        int sec = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_MEMBERSHIP)
+                .columns(CLANS_CLAN_MEMBERSHIP.CLAN_ID,CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .values(this.clanID,receiver.getUserId())
+                .execute();
+
+        if (sec != 1) throw new IllegalStateException("Inserted gg.solarmc.loader.clans.ClanMember already belongs to clan!");
```

Comment:
There are 2 possibilities for your INSERT statement:
- The insert will succeed, and 1 row will be inserted, and `sec` will thus be 1 (sec is the update count)
- The insert will fail due to a constraint violation, and you will receive an exception

So it is not possible for `sec != 1`

Reply:
How do i check for constraint violation (what exception is thrown)?

Reply:
Oh my god i didn't read my own code - closed!